webapp/src/pages/JoinChatroomPage/JoinChatroomPage.tsx

```diff
@@ -0,0 +1,131 @@
+import { useCallback, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import {
+  Button,
+  ButtonVariant,
+  Header,
+  IconVariant,
+  Input,
+  InputVariant,
+  LargeAvatar,
+  Loading,
+  Text,
+  TextVariant,
+} from '../../shared/components';
+import { useNotificationContext } from '../../shared/context/NotificationContext';
+import { ResponseError } from '../../shared/generated';
+import { useData } from '../../shared/hooks/UseData';
+import { useNavigation } from '../../shared/hooks/UseNavigation';
+import { chatApi } from '../../shared/services/ApiService';
+import {
+  CHATROOM_EP_URL,
+  CHATROOM_URL,
+  WILDCARD_AVATAR_URL,
+} from '../../shared/urls';
+import NotFoundPage from '../NotFoundPage/NotFoundPage';
+import './JoinChatroomPage.css';
+
+export default function JoinChatroomPage() {
+  const { chatroomId } = useParams();
+
+  if (!chatroomId) {
+    return null;
+  }
+  return <JoinChatroom chatroomId={chatroomId} />;
+}
+
+type JoinChatroomProps = {
+  chatroomId: string;
+};
+
+function JoinChatroom({ chatroomId }: JoinChatroomProps) {
+  const { goBack } = useNavigation();
+  const getChatroom = useCallback(
+    () => chatApi.chatControllerGetChatroomById({ id: chatroomId }),
+    [chatroomId],
+  );
+  const { data: chatroom, isLoading: isChatroomLoading } = useData(getChatroom);
+  const isProtectedChatroom = chatroom?._public === false;
+  const [password, setPassword] = useState('');
+  const { warn, notify } = useNotificationContext();
+  const { navigate } = useNavigation();
+
+  const handleSubmit: React.FormEventHandler<HTMLFormElement> = async (
+    event,
+  ) => {
+    event.preventDefault();
+    try {
+      await chatApi.chatControllerCreateChatroomMember({
+        chatroomId: chatroomId,
+        joinChatroomDto: {
+          password: isProtectedChatroom ? password : undefined,
+        },
+      });
+      notify('You joined a chatroom');
+      navigate(`${CHATROOM_URL}/${chatroomId}`, { replace: true });
+    } catch (error: unknown) {
+      if (error instanceof ResponseError) {
+        const responseBody = await error.response.json();
+        if (responseBody.message) {
+          warn(responseBody.message);
+        } else {
+          warn(error.response.statusText);
+        }
+      } else if (error instanceof Error) {
+        warn(error.message);
+      } else {
+        warn('Could not join the chatroom');
+      }
+    }
+  };
+
+  if (isChatroomLoading) {
+    return (
+      <div className="join-chatroom">
+        <div className="join-chatroom-loading">
+          <Loading />
+        </div>
+      </div>
+    );
+  }
+
+  if (!chatroom) {
+    return <NotFoundPage />;
+  }
+
+  return (
+    <div className="join-chatroom">
+      <Header icon={IconVariant.ARROW_BACK} onClick={goBack}>
+        join chatroom
+      </Header>
+      <div className="join-chatroom-avatar">
+        <LargeAvatar
+          url={
+            chatroom.avatarId
```

Comment:
Also, are we sure it's reasonable to make different endpoints for avatars for chat and users?
In `webapp/src/pages/ChatPage/ChatPage.tsx` line 28, we are presuming it's `url: `${AVATAR_EP_URL}/${chatroom.avatarId}`,` instead

Reply:
Good catch! I've changed the `url` of the avatar in the `ChatPage`.

I think we need a different endpoint for the chatroom avatars because the current endpoint is for the user avatars `/api/v1/users/avatars/{avatarId}`, or we could create a new controller for the avatars with an endpoint like `/api/v1/avatar/{avatarId}`

Reply:
I think the new controller with the `/api/v1/avatar/{avatarId}` endpoint would be the way to go. But it's not for this PR, obviouly.